src/legacy/core_plugins/table_vis/migrations.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { cloneDeep, get, omit } from 'lodash';
+
+export const migrations = {
+  visualization: {
+    '7.0.0': (doc) => {
+      try {
+        const visState = JSON.parse(doc.attributes.visState);
+        if (get(visState, 'type') !== 'table') {
+          return doc; // do nothing; we only want to touch tables
+        }
+
+        let splitCount = 0;
+        visState.aggs = visState.aggs.map(agg => {
+          if (agg.schema === 'split') {
+            splitCount++;
+            if (splitCount === 1) {
+              return agg; // leave the first split agg unchanged
+            }
+            agg.schema = 'bucket';
+            // the `row` param is exclusively used by split aggs, so we remove it
+            agg.params = omit(agg.params, ['row']);
+          }
+          return agg;
+        });
+
+        if (splitCount <= 1) {
+          return doc; // do nothing; we only want to touch tables with multiple split aggs
+        }
+
+        const newDoc = cloneDeep(doc);
+        newDoc.attributes.visState = JSON.stringify(visState);
+        return newDoc;
+      }
+      catch (e) {
+        // if anything goes wrong, do nothing and move on
```

Comment:
Is this the right thing to do? It'll leave you with old visualizations which are now considered 7.0.0 by the migration system. I think it's either better to fail or at least have some property that you add to the doc that indicates it's failed so that we can later clean them up or migrate them once we figure out how. (e.g. `return { ...doc, attributes: { ...doc.attributes, migrationFailed: '7.0.0' } }` or something.)

Reply:
@chrisdavies I guess I should've investigated further beforehand, but come to think of it, I'm not sure I know what happens if I throw an error here mid-migration. I assume it aborts and reverts to the previously backed-up index?

Reply:
Throwing an error will cause Kibana to fail to start, so it's not great, either. I think failure cases are currently a relatively unsolved problem. :/ I'm happy to hop on Zoom or whatever to have a discussion.